### PR TITLE
Add support for ARM64

### PR DESCRIPTION
**Pull request by bastianeicher**

### CODE REVIEW

File: src/zeroinstall/arch.ml [hunk 1]
```diff
@@ -72,7 +72,10 @@ let get_machine_ranks ~multiarch machine =
     | "ppc"     -> [|"ppc32"|]
     | "x86_64" when multiarch -> [|"i686"; "i586"; "i486"; "i386"|]
     | "ppc64"  when multiarch -> [|"ppc"|]
-    | "armv7l" -> [| "armv6l" |]
+    | "armv7l"  -> [| "armv6l" |]
+    | "armhf"   -> [| "armv7l" |]
```
Comment (talex5) [inline src/zeroinstall/arch.ml:76]: I guess this needs to list `armv6l` too.

File: src/zeroinstall/arch.ml [hunk 2]
```diff
@@ -72,7 +72,10 @@ let get_machine_ranks ~multiarch machine =
...
+    | "aarch64" -> [| "arm64" |]
+    | "aarch64"  when multiarch -> [|"armv6l"; "armv7l"|]
```
Comment (talex5) [inline src/zeroinstall/arch.ml:78]: These lines need to be the other way around. Also, the multiarch case presumably should include `arm64`.

**Review by talex5 — CHANGES_REQUESTED** [review]

File: src/zeroinstall/arch.ml (hunk 1 above)
Comment (bastianeicher) [reply src/zeroinstall/arch.ml]: That's right. I've now added it.

File: src/zeroinstall/arch.ml (hunk 2 above)
Comment (bastianeicher) [reply src/zeroinstall/arch.ml]: I've restructured it like this now:

```ocaml
| "aarch64" when multiarch -> [|"armv7l"; "armv6l"|]
| "arm64"   when multiarch -> [|"aarch64"; "armv7l"; "armv6l"|]
| "arm64" -> [|"aarch64"|]
```

What I'm trying to express is that for ARM64 machines:

- ... feeds are always expected to list the CPU type as `aarch64`.
- ... the local OS / `uname` may report the CPU as `amd64` or `aarch64`.
- ... the system can run also run ARMv6 and ARMv7 code when in 32-bit mode.

Comment (talex5) [reply src/zeroinstall/arch.ml]: Ah, probably the `arm64` thing should be added here instead then:

https://github.com/0install/0install/blob/5ea4b65e1eaf4719d4ec26170bcfff8d2497edb0/src/support/system.ml#L52-L63

Comment (bastianeicher) [reply src/zeroinstall/arch.ml]: Ah, indeed! I'd completely overlooked `canonical_machines`. Adjusted the PR accordingly.

Perhaps `ppc` -> `ppc32` should also be moved here?

**Review by talex5 — APPROVED** [review]

**Pull request merged**